### PR TITLE
Teams-Fx Toolkit Migration Nodejs

### PR DESCRIPTION
**Pull request by Mohammed-MSFT**

### CODE REVIEW

File: samples/meetings-notification/nodejs/views/index.html [hunk 1]
```diff
@@ -26,7 +26,7 @@
                     myValue: document.forms["contentForm"]["option"].value,
                     title: "<%= question %>"
                 }
-                microsoftTeams.tasks.submitTask(contentInfo, "<<APP_ID>>");
+                microsoftTeams.tasks.submitTask(contentInfo, "daecec22-636b-470f-ab26-c60de025cde6");
```
Comment (Prithvi-MSFT) [inline samples/meetings-notification/nodejs/views/index.html:29]: remove this id, it should not come from client side

File: samples/tab-request-approval/nodejs/server.js [hunk 2]
```diff
@@ -1,25 +1,38 @@
 const express = require('express');
-const bodyparser = require('body-parser');
-const env = require('dotenv')
+// const bodyparser = require('body-parser');
```
Comment (Prithvi-MSFT) [inline samples/tab-request-approval/nodejs/server.js:2]: remove commented code

File: samples/tab-request-approval/nodejs/views/UserNotification.ejs [hunk 3]
```diff
@@ -34,7 +34,7 @@
             <div id="createtask" class="tab-pane fade in active">
                 <div class="mb">
                     <div class="form-group fg">
-                        <mgt-teams-msal2-provider client-id="9811a0fb-ea05-46c7-b631-9c24f55ffe0c"
+                        <mgt-teams-msal2-provider client-id="df99fae2-35ed-46a7-b946-c1b42ae885fb"
```
Comment (Prithvi-MSFT) [inline samples/tab-request-approval/nodejs/views/UserNotification.ejs:37]: remove client id. It should come from server only

**Review by Prithvi-MSFT — COMMENTED** [review]
provided review comments

File: samples/meetings-notification/nodejs/views/index.html (hunk 1 above)
Comment (Mohammed-MSFT) [reply samples/meetings-notification/nodejs/views/index.html:29]: fixed

File: samples/tab-request-approval/nodejs/server.js (hunk 2 above)
Comment (Mohammed-MSFT) [reply samples/tab-request-approval/nodejs/server.js:2]: fixed

File: samples/tab-request-approval/nodejs/views/UserNotification.ejs (hunk 3 above)
Comment (Mohammed-MSFT) [reply samples/tab-request-approval/nodejs/views/UserNotification.ejs]: fixed

File: samples/outgoing-webhook/nodejs/.env
Comment (formulahendry) [inline samples/outgoing-webhook/nodejs/.env:1]: remove this file

File: samples/outgoing-webhook/nodejs/teamsapp.local.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      teamsAppId: TEAMS_APP_ID
+
+  - uses: script
+    with:
+      run:
+        # echo "::set-teamsfx-env MICROSOFT_APP_TYPE=SingleTenant";
+        echo "::set-teamsfx-env MICROSOFT_APP_TYPE=MultiTenant";
+        echo "::set-teamsfx-env MICROSOFT_APP_TENANT_ID=${{AAD_APP_TENANT_ID}}";
+
+  - uses: arm/deploy # Deploy given ARM templates parallelly.
```
Comment (formulahendry) [inline samples/outgoing-webhook/nodejs/teamsapp.local.yml:37]: Does this sample need to create bot?

File: samples/outgoing-webhook/nodejs/README.md [hunk 5]
```diff
@@ -29,6 +29,19 @@ This project is an implementation of a very simple [Outgoing Webhook](https://do
...
+
+## Run the app (Using Teams Toolkit for Visual Studio Code)
+
+The simplest way to run this sample in Teams is to use Teams Toolkit for Visual Studio Code.
+
+1. Ensure you have downloaded and installed [Visual Studio Code](https://code.visualstudio.com/docs/setup/setup-overview)
+1. Install the [Teams Toolkit extension](https://marketplace.visualstudio.com/items?itemName=TeamsDevApp.ms-teams-vscode-extension)
+1. Select **File > Open Folder** in VS Code and choose this samples directory from the repo
+1. Using the extension, sign in with your Microsoft 365 account where you have permissions to upload custom apps
+1. Select **Debug > Start Debugging** or **F5** to run the app in a Teams web client.
```
Comment (formulahendry) [inline samples/outgoing-webhook/nodejs/README.md:42]: For this webhook sample, I think current steps would not work?

File: samples/tab-request-approval/nodejs/server.js [hunk 6]
```diff
@@ -103,8 +105,8 @@ app.post('/auth/token', function (req, res) {
       method: "POST",
       body: querystring.stringify(params),
       headers: {
-        Accept: "application/json",
-        "Content-Type": "application/x-www-form-urlencoded"
+        Accept: "serverlication/json",
```
Comment (formulahendry) [inline samples/tab-request-approval/nodejs/server.js:108]: typo?

File: samples/outgoing-webhook/nodejs/.env
Comment (Mohammed-MSFT) [reply samples/outgoing-webhook/nodejs/.env:1]: fixed

File: samples/tab-request-approval/nodejs/server.js (hunk 6 above)
Comment (Mohammed-MSFT) [reply samples/tab-request-approval/nodejs/server.js:108]: fixed

File: samples/outgoing-webhook/nodejs/README.md (hunk 5 above)
Comment (Mohammed-MSFT) [reply samples/outgoing-webhook/nodejs/README.md:42]: For each sample we were updating readme with common toolkit migration steps.
Could you please let us know what's need to be added.

File: samples/outgoing-webhook/nodejs/teamsapp.local.yml (hunk 4 above)
Comment (Mohammed-MSFT) [reply samples/outgoing-webhook/nodejs/teamsapp.local.yml:37]: yeah, using bot formula its works for us, some samples are not working with tab formula...so we are switching for bot formula.
Provide your input please.
Thanks
MicrosoftTeams-image (1)

Comment (formulahendry) [reply samples/outgoing-webhook/nodejs/teamsapp.local.yml:37]: So you need to remove bot related things. e.g. ` arm/deploy`

Comment (formulahendry) [reply samples/outgoing-webhook/nodejs/teamsapp.local.yml:37]: I think the whole teamsapp.local.yml is not needed for this sample

File: samples/outgoing-webhook/nodejs/README.md (hunk 5 above)
Comment (formulahendry) [reply samples/outgoing-webhook/nodejs/README.md:42]: There is a step to create webhook. Without that, it would not work.

Comment (Mohammed-MSFT) [reply samples/outgoing-webhook/nodejs/README.md:42]: Ok

**Review by formulahendry — APPROVED** [review]

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Comment by ChetanSharma-msft** [comment]
@Mohammed-MSFT - Please run build pipeline before merging this PR.

File: samples/outgoing-webhook/nodejs/teamsapp.local.yml (hunk 4 above)
Comment (ChetanSharma-msft) [reply samples/outgoing-webhook/nodejs/teamsapp.local.yml]: Removed it.

File: samples/outgoing-webhook/nodejs/README.md (hunk 5 above)
Comment (ChetanSharma-msft) [reply samples/outgoing-webhook/nodejs/README.md]: Updated to normal sample.

**Comment by ChetanSharma-msft** [comment]
@Mohammed-MSFT - Please check failed build pipeline.

**Comment by Prithvi-MSFT** [comment]
Build status - All builds passed
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/5710770366

**Review by Prithvi-MSFT — APPROVED** [review]

**Pull request merged**